torch/csrc/jit/script/ir_emitter.cpp

```diff
@@ -1797,22 +1797,36 @@ struct to_ir {
   // If the RHS is a tensor, return the corresponding ATen in-place op
   // If it's a list of scalars, then return the corresponding list augment op
   Symbol getAugOp(const AugAssign& stmt, const TypePtr& type) {
-    if (type->cast<ListType>()) { // Lists also have in-place ops.
-      switch (stmt.aug_op()) {
-        case '+':
-          return aten::add_;
-      }
+    bool use_inplace_op = type->isSubtypeOf(TensorType::get()) ||
+        type->kind() == TypeKind::ListType;
+    switch (stmt.aug_op()) {
+      case '+':
+        return use_inplace_op ? aten::add_ : aten::add;
+      case '-':
+        return use_inplace_op ? aten::sub_ : aten::sub;
+      case '/':
+        return use_inplace_op ? aten::div_ : aten::div;
+      case '*':
+        return use_inplace_op ? aten::mul_ : aten::mul;
+      default:
+        throw ErrorReport(stmt)
+            << "Unknown augmented assignment: " << kindToString(stmt.aug_op());
     }
-    bool isTensor = type->isSubtypeOf(TensorType::get());
+  }
+
+  // Get a pair of <in place magic method name, out of place magic method name>
+  // since the out of place method is called if the in place method is not
+  // present
+  std::pair<const char*, const char*> getAugMagicMethod(const AugAssign& stmt) {
```

Comment:
apparently all string literals are permanently allocated, so this is memory safe. but if someone were to edit this in a way that dynamically creates string literals it no longer would be. since these `const char *` can upcasted in `makeMagic` to `std::string` anyway can you use `std::string` here instead of `const char *`? It is also one less thing to think about when reviewing and maintaining this code.